relion/protocols/protocol_compress_movies.py

```diff
@@ -0,0 +1,165 @@
+        return None if ext in ['.mrc', '.mrcs', '.tiff', '.tif'] else 'mrc'
+
+    # -------------------------- DEFINE param functions -----------------------
+    def _defineAlignmentParams(self, form):
+
+        form.addParam('inputGainProt', params.PointerParam, allowsNull=True,
+                      pointerClass='ProtRelionCompressEstimateGain',
+                      label='Input gain (Optional)',
+                      help='Provide as input a compress estimate protocol '
+                           'from where the estimated gain file will be taken.')
+
+        group = form.addGroup("TIFF Options")
+
+        group.addParam('compression', params.EnumParam, default=1,
+                       choices=['none', 'auto', 'zip', 'lzw'],
+                       label='Compression type')
+
+        group.addParam('deflateLevel', params.IntParam, default=6,
+                       label='Deflate level',
+                       help="deflate level. 1 (fast) "
+                            "to 9 (slowest but best compression)")
+
+        form.addParallelSection(threads=4, mpi=1)
+
+    # --------------------------- STEPS functions -------------------------------
+    def _convertInputStep(self):
+        self.info("Relion version:")
+        self.runJob("relion_convert_to_tiff --version", "", numberOfMpi=1)
+        self.info("Detected version from config: %s"
+                  % relion.Plugin.getActiveVersion())
+        # Create a local link to the input gain file if necessary
+        inputGain = self.getInputGain()
+        if inputGain:
+            tmpGain = self._getTmpPath('gain_estimate.bin')
+            pwutils.createLink(inputGain, tmpGain)
+            pwutils.createLink(inputGain.replace('.bin', '_reliablity.bin'),
+                               tmpGain.replace('.bin', '_reliablity.bin'))
+        ProtAlignMovies._convertInputStep(self)
+
+    def _processMovie(self, movie):
+        fn = movie.getAttributeValue('_originalFileName', movie.getFileName())
+        baseName = os.path.basename(fn)
+        compression = self.getEnumText('compression')
+        pwutils.createLink(fn, self._getTmpPath(baseName))
+        args = "--i %s --o ../extra/ " % baseName
+        args += "--compression %s " % compression
+        # TODO: Check if deflateLevel is only valid for zip (deflate)
+        if compression == 'zip':  # deflate
+            args += "--deflate_level %d" % self.deflateLevel
+
+        if self.getInputGain():
+            args += "--gain gain_estimate.bin "
+
+        self.runJob('relion_convert_to_tiff', args, cwd=self._getTmpPath())
+
+    # --------------------------- INFO functions ------------------------------
+    def _summary(self):
+        summary = []
+        return summary
+
+    def _citations(self):
+        return ['Zivanov2019']
+
+    def _validate(self):
+        # Check base validation before the specific ones for Motioncor
+        errors = ProtAlignMovies._validate(self)
```

Comment:
why alignmovies here?

Reply:
Despite it is not purely an "AlignMovies" protocol, it has some similarities. I mainly used it to re-use the streaming infrastructure.